pkg/services/alerting/rule.go

```diff
@@ -89,7 +89,7 @@ func getTimeDurationStringToSeconds(str string) (int64, error) {
 
 	matches := valueFormatRegex.FindAllString(str, 1)
 
-	if len(matches) <= 0 {
+	if len(matches) == 0 {
```

Comment:
From the [FindAllString()](https://golang.org/pkg/regexp/#Regexp.FindAllString) docs: `A return value of nil indicates no match.`
so maybe this should check if the `matches` is `nil` instead.

Reply:
@papagian I think it's actually considered OK to check the length of a nil slice in Go, see [this example](https://tour.golang.org/moretypes/12#:~:text=The%20zero%20value%20of%20a,and%20has%20no%20underlying%20array.&text=fmt.,-Println(%22nil!). 

We can check against `nil` instead, but I don't think there's any requirement calling for it :)

Reply:
yes, let's keep it as is.